network/topics/pubsub.go

```diff
@@ -108,7 +109,7 @@ func (cfg *PubSubConfig) initScoring() {
 }
 
 // NewPubSub creates a new pubsub router and the necessary components
-func NewPubSub(ctx context.Context, logger *zap.Logger, cfg *PubSubConfig, metrics Metrics) (*pubsub.PubSub, Controller, error) {
+func NewPubSub(ctx context.Context, logger *zap.Logger, cfg *PubSubConfig, metrics Metrics, getCommittees func() []*storage.Committee) (*pubsub.PubSub, Controller, error) {
```

Comment:
if you define an interface here for just that method it's cleaner and also on the caller side you can just pass the ValidatorStore as it is (without declaring an anonymous func)
example:
```suggestion
type CommitteesProvider interface {
    Committees() []*storage.Committee
}

// NewPubSub creates a new pubsub router and the necessary components
func NewPubSub(ctx context.Context, logger *zap.Logger, cfg *PubSubConfig, metrics Metrics, committeesProvider CommitteesProvider) (*pubsub.PubSub, Controller, error) {
```

Reply:
Awesome, thanks!